filament/backend/src/metal/MetalDriver.h

```diff
@@ -112,8 +112,9 @@ class MetalDriver final : public DriverBase {
 
     template<typename Dp, typename B>
     Dp* handle_cast(HandleMap& handleMap, Handle<B> handle) noexcept {
-        std::lock_guard<std::mutex> lock(mHandleMapMutex);
         assert(handle);
+        if (!handle) return nullptr; // better to get a NPE than random behavior/corruption
+        std::lock_guard<std::mutex> lock(mHandleMapMutex);
```

Comment:
Probably worth adding this to `handle_const_cast` and `destruct_handle` as well (same for Vulkan driver).